corehq/apps/api/tests/test_case_update_api.py

```diff
@@ -0,0 +1,179 @@
+            '@case_type': 'player',
+            '@case_name': 'Elizabeth Harmon',
+            '@owner_id': 'methuen_home',
+            'properties': {
+                'external_id': '1',
+                'sport': 'chess',
+                'dob': '1948-11-02',
+            },
+        })
+        self.assertEqual(res.keys(), ['@case_id', '@form_id'])
+        case = self.case_accessor.get_case(res['@case_id'])
+        self.assertEqual(case.domain, self.domain)
+        self.assertEqual(case.type, 'player')
+        self.assertEqual(case.name, 'Elizabeth Harmon')
+        self.assertEqual(case.owner_id, 'methuen_home')
+        self.assertEqual(case.opened_by, 'netflix')
+        self.assertEqual(case.external_id, '1')
+        self.assertEqual(case.dynamic_case_properties(), {
+            'dob': '1948-11-02',
+            'sport': 'chess',
+        })
+
+        xform = self.form_accessor.get_form(res['@form_id'])
+        self.assertEqual(xform.metadata.xmlns, 'http://commcarehq.org/case_api')
+        self.assertEqual(xform.metadata.userID, self.web_user.user_id)
+        self.assertEqual(xform.metadata.deviceID, 'user agent string')
+
+    def _make_case(self):
+        xform, cases = submit_case_blocks([CaseBlock(
+            case_id=str(uuid.uuid4()),
+            case_type='player',
+            case_name='Elizabeth Harmon',
+            owner_id='methuen_home',
+            create=True,
+            update={
+                'external_id': '1',
+                'sport': 'chess',
+                'rank': '1600',
+                'dob': '1948-11-02',
+            }
+        ).as_text()], domain=self.domain)
+        return cases[0]
+
+    def test_update_case(self):
+        case = self._make_case()
+
+        res = self.post(f'update_case/{case.case_id}', {
+            # notable exclusions: case_id, date_opened, date_modified, case_type
+            '@case_name': 'Beth Harmon',
+            '@owner_id': 'us_chess_federation',
+            'properties': {
+                'rank': '2100',
+                'champion': 'true',
+            },
+        })
+        self.assertEqual(res.keys(), ['@case_id', '@form_id'])
+
+        case = self.case_accessor.get_case(case.case_id)
+        self.assertEqual(case.name, 'Beth Harmon')
+        self.assertEqual(case.owner_id, 'us_chess_federation')
+        self.assertEqual(case.dynamic_case_properties(), {
+            'champion': 'true',
+            'dob': '1948-11-02',
+            'rank': '2100',
+            'sport': 'chess',
+        })
+
+    def test_create_child_case(self):
+        parent_case = self._make_case()
+        res = self.post('create_case', {
+            '@case_type': 'match',
+            '@case_name': 'Harmon/Luchenko',
+            '@owner_id': 'harmon',
+            'properties': {
+                'external_id': '23',
+                'winner': 'Harmon',
+            },
+            'indices': {
+                'parent': {
```

Comment:
I don't think we have standards in HQ for how index properties are called, but this is generally in-line with mobile's [casexml spec](https://github.com/dimagi/commcare-core/wiki/casexml20)